core/src/display_object/avm1_button.rs

```diff
@@ -195,31 +193,36 @@ impl<'gc> Avm1Button<'gc> {
         }
     }
 
-    pub fn enabled(self, context: &mut UpdateContext<'_, 'gc>) -> bool {
-        if let Some(object) = self.0.read().object {
+    fn get_property_as_bool(
```

Comment:
nit: How about renaming this function to `get_boolean_property`? (Same applies to `movie_clip.rs`)
Also I suggest renaming the `is_undefined_truthy` parameter to `default` - IMO it better reflects its meanings.

Reply:
Good idea, I've done the changes. Also, thank you for the review.